packages/astro/src/assets/utils/node/emitAsset.ts

```diff
@@ -44,7 +45,7 @@ export async function emitESMImage(
 	});
 
 	// Attach file data for SVGs
-	if (fileMetadata.format === 'svg') {
+	if (fileMetadata.format === 'svg' && experimentalSvgEnabled === true) {
```

Comment:
Could you add a TODO with the proper solution here? (or at least just "figure a better solution") We definitely need to fix it before leaving experimental.

A good fix would be to make this lazy.. but it's unfortunately annoying in serverless (it'd need to work like the image endpoint with a fetch, which is doable, honestly) + having it lazy would mean that this could also exists for non-svg, a feature people have been asking for since forever